src/httpserver.cpp

```diff
@@ -634,7 +634,7 @@ void HTTPRequest::WriteHeader(const std::string& hdr, const std::string& value)
  * Replies must be sent in the main loop in the main http thread,
  * this cannot be done from worker threads.
  */
-void HTTPRequest::WriteReply(int nStatus, const std::string& strReply)
+void HTTPRequest::WriteReply(int nStatus, const std::string_view strReply)
```

Comment:
Not sure if a `Span<byte>` or string_view is best suited here.  In principle we're dealing with binary data here, not necessarily a string.

Reply:
Thanks for the suggestion!

What do you think about the following modification, in order to support `Span`?
```diff
diff --git a/src/httpserver.cpp b/src/httpserver.cpp
index 54fec59468..adc6040d08 100644
--- a/src/httpserver.cpp
+++ b/src/httpserver.cpp
@@ -634,7 +634,7 @@ void HTTPRequest::WriteHeader(const std::string& hdr, const std::string& value)
  * Replies must be sent in the main loop in the main http thread,
  * this cannot be done from worker threads.
  */
-void HTTPRequest::WriteReply(int nStatus, const std::string_view strReply)
+void HTTPRequest::WriteReply(int nStatus, Span<const std::byte> reply)
 {
     assert(!replySent && req);
     if (m_interrupt) {
@@ -643,7 +643,7 @@ void HTTPRequest::WriteReply(int nStatus, const std::string_view strReply)
     // Send event to main http thread to send reply message
     struct evbuffer* evb = evhttp_request_get_output_buffer(req);
     assert(evb);
-    evbuffer_add(evb, strReply.data(), strReply.size());
+    evbuffer_add(evb, reply.data(), reply.size());
     auto req_copy = req;
     HTTPEvent* ev = new HTTPEvent(eventBase, true, [req_copy, nStatus]{
         evhttp_send_reply(req_copy, nStatus, nullptr, nullptr);
diff --git a/src/httpserver.h b/src/httpserver.h
index a452465636..1d43c5f3b3 100644
--- a/src/httpserver.h
+++ b/src/httpserver.h
@@ -9,6 +9,8 @@
 #include <optional>
 #include <string>
 
+#include <span.h>
+
 namespace util {
 class SignalInterrupt;
 } // namespace util
@@ -123,12 +125,16 @@ public:
     /**
      * Write HTTP reply.
      * nStatus is the HTTP status code to send.
-     * strReply is the body of the reply. Keep it empty to send a standard message.
+     * reply is the body of the reply. Keep it empty to send a standard message.
      *
      * @note Can be called only once. As this will give the request back to the
      * main thread, do not call any other HTTPRequest methods after calling this.
      */
-    void WriteReply(int nStatus, const std::string_view strReply = "");
+    void WriteReply(int nStatus, std::string_view reply = "") 
+    {
+        WriteReply(nStatus, AsBytes(Span{reply}));
+    }
+    void WriteReply(int nStatus, Span<const std::byte> reply);
 };
 
 /** Get the query parameter value from request uri for a specified key, or std::nullopt if the key
```

Reply:
Added https://github.com/bitcoin/bitcoin/commit/9c9375cf3b581c31f047b87a2c05507dc7b31804 with a few more fixes.